### PR TITLE
Added a configurable limit to the number of groups a user can create …

### LINKED ISSUE #97: Limit number of groups a user can create
A user may only create 5 groups. 
A user can only be part of 5 groups
Make this limit configurable.

### PR DESCRIPTION
**Pull request by VladCuciureanu**
### What does it fix?

Closes #97 

This PR implements configurable limits to how many groups can be created by an user, and also for how many groups can be joined by an user.

Also renamed a model since it's name contained a typo 😊

### How has it been tested?

Wrote a test for each test case in which the limit is exceeded. If tests should be written for the cases where the limit is not exceeded, please contact me :)

### CODE REVIEW

File: Src/Tests/DeUrgenta.Group.Api.Tests/CommandsHandlers/AddGroupHandlerShould.cs [hunk 1]
```diff
@@ -16,10 +22,50 @@ namespace DeUrgenta.Group.Api.Tests.CommandsHandlers
...
+
+            // Seed user
+            var user = _dbContext.Users.AddAsync(new User {FirstName = "test", LastName = "test", Sub = "a-sub"})
+                .Result.Entity;
+
+            // Seed groups
+            var groups = new List<Domain.Entities.Group>();
+            for (int i = 0; i < 5; i++)
+            {
+                groups.Add(new Domain.Entities.Group {AdminId = user.Id, Admin = user, Name = "TestGroup"});
```
Comment (idormenco) [inline Src/Tests/DeUrgenta.Group.Api.Tests/CommandsHandlers/AddGroupHandlerShould.cs]: this assignment `AdminId = user.Id` can be removed. (i think user.id is an empty guid here ,but I might be wrong)

File: Src/DeUrgenta.Group.Api/CommandHandlers/AddGroupHandler.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@
 using DeUrgenta.Group.Api.Models;
 using MediatR;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
```
Comment (idormenco) [inline Src/DeUrgenta.Group.Api/CommandHandlers/AddGroupHandler.cs]: remove unused import

File: Src/DeUrgenta.Group.Api/CommandHandlers/InviteToGroupHandler.cs [hunk 3]
```diff
@@ -1,12 +1,15 @@
...
 using System.Threading.Tasks;
 using CSharpFunctionalExtensions;
 using DeUrgenta.Common.Validation;
 using DeUrgenta.Domain;
 using DeUrgenta.Domain.Entities;
 using DeUrgenta.Group.Api.Commands;
 using MediatR;
+using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
```
Comment (idormenco) [inline Src/DeUrgenta.Group.Api/CommandHandlers/InviteToGroupHandler.cs]: remove unused import

**Review by idormenco — CHANGES_REQUESTED** [review]

File: Src/Tests/DeUrgenta.Group.Api.Tests/CommandsHandlers/AddGroupHandlerShould.cs (hunk 1 above)
Comment (VladCuciureanu) [reply Src/Tests/DeUrgenta.Group.Api.Tests/CommandsHandlers/AddGroupHandlerShould.cs]: Not having an AdminId or an Admin breaks a foreign key. Should I try it again?

Comment (idormenco) [reply Src/Tests/DeUrgenta.Group.Api.Tests/CommandsHandlers/AddGroupHandlerShould.cs]: you should keep only Admin

**Review by idormenco — APPROVED** [review]
lgtm

**Pull request merged**